ci: Fix and improve tests workflow (fix: #15380)

yusufkandemir opened the pull request:
**What kind of change does this PR introduce?**

- CI

**Does this PR introduce a breaking change?**

- No

**The PR fulfills these requirements:**

- It's submitted to the `dev` branch (or `v[X]` branch)
- When resolving a specific issue, it's referenced in the PR's title (e.g. `fix: #xxx[,#xxx]`, where "xxx" is the issue number)

**Other information:**
Resolves #15380

We decided to drop Cypress Cloud on PR runs. It's either a security risk or doesn't make sense to have workflow at all depending on which path is used, see the linked issue for some info. Cypress doesn't offer a way to output Cypress Cloud-ready results into a file, so we can't use the `pull_request` + `workflow_run` pattern. So, the only feasible way left is using `pull_request` + `workflow_run` but dropping Cypress Cloud, using custom reporting instead. We lose other benefits of Cypress Cloud, but this is about the best we can do right now. If Cypress starts offering a way to save/upload Cypress Cloud-ready results, we can adjust the workflows to use it.

yusufkandemir committed:
ci: Use yarn in tests workflow

yusufkandemir committed:
ci: Improve caching in tests workflow

yusufkandemir committed:
ci: Correctly pass browser in tests workflow
Lots of parameters including 'browser' is disabled when using the 'command' parameter
https://github.com/cypress-io/github-action#custom-test-command

yusufkandemir committed:
ci: Run tests workflow in PR code, drop Cypress Cloud

Side note: 'tag' and 'group' wasn't being used due to 'command' usage
https://github.com/cypress-io/github-action#custom-test-command

yusufkandemir committed:
ci: Improve filtered paths in tests workflow

yusufkandemir committed:
ci: Add tests result reporting

yusufkandemir commented:
This PR should be rebased after merging #15766 in, only after then it could get merged.

yusufkandemir committed:
ci: Do not use yarn install --focus to avoid errors
--focus requires all packages to be published for some reason,
but the vite-plugin playground is not.

pnpm --filter XYZ would have been useful here.

I have used this to save some time by avoiding to install packages for
other workspaces. But, it should not be a big deal now that we are
utilizing caching.

yusufkandemir commented:
The workflows are failing due to a separate problem in the build step: https://github.com/quasarframework/quasar/commit/5fe8dcf4d925543d02be1db39ae3f78e1283d0b1#commitcomment-110857726
Easy to fix, but I don't want to risk a potential conflict as Raz might have fixed it but forgot to push it. This PR is not ready to merge anyways, so not a problem.

Hopefully, the updated workflows will be working fine. When they pass, they should also post a comment on the PR with the test results.

IlCallo commented:
Seems like Raz added the missing JSON API definitions here: https://github.com/quasarframework/quasar/commit/cea8485ec9d24b0b42db322f60e05233285c2288

yusufkandemir committed:
ci: Use Node v16 in tests workflow

yusufkandemir committed:
chore: Update Node version to v16 in contribution guide

yusufkandemir committed:
ci: Use Node v16 in tests workflow properly

yusufkandemir committed:
ci: Correctly pass reporter options to test run

yusufkandemir committed:
chore(ui): Bump Quasar dev deps

yusufkandemir committed:
ci: fix/improve test JSON reporting
outputting to a file doesn't work as advertised, on top of that,
https://glebbahmutov.com/blog/json-test-results/ mentions that
using json reporter with output disables the usual terminal output

yusufkandemir committed:
ci: Build vite-plugin package as well
Yarn workspaces link everything by default, so we need to build vite-plugin as well
Unlike Yarn, pnpm allows opt-in with "workspace:" protocol, so would have
been better here.
